homeassistant/components/zha/config_flow.py

```diff
@@ -0,0 +1,55 @@
+"""Config flow for ZHA."""
+from collections import OrderedDict
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+
+from .const import (
+    DOMAIN, BAUD_RATES, CONF_BAUDRATE, CONF_DATABASE, CONF_RADIO_TYPE,
+    CONF_USB_PATH, RadioType
+)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class ZhaFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_user(self, user_input=None):
+        """Handle a zha config flow start."""
+        if self._async_current_entries():
+            return self.async_abort(reason='single_instance_allowed')
+
+        errors = {}
+
+        if user_input is not None:
+            return self.async_create_entry(
+                title=user_input[CONF_USB_PATH], data=user_input)
+
+        fields = OrderedDict()
+        fields[vol.Required(CONF_DATABASE)] = str
+        fields[vol.Required(CONF_USB_PATH)] = str
```

Comment:
Can we detect this automatically?

"We found a X stick by Y. Is this the one you want to set up?"

Reply:
I think in this case we just should ask just for the serial port path and radio type via configuration entry. Zigbee db path -- hardcoded, knowing the radio type we can choose the default baudrate (or use 57600 for both ezsp and xbee radios) and at the moment I cannot think of a good cross-platform way of detecting all available serial ports.

Reply:
Can we use pyserial list_ports tool?

https://pyserial.readthedocs.io/en/latest/tools.html

Reply:
I think we can use serial.tools.list_ports and maybe even prefer a specific pid/vid on top of the list. In my environment both HUSBZB-1 have `VID:PID=10C4:8A2A` Usually the zha radio is going to be on `/dev/tty/USB1` but if you have any other usb serial ports then it may shift.

Reply:
Let's keep that in mind for a future PR (or even auto discovery of this config entry based on available sticks???)